.github/workflows/build-and-deploy.yml

```diff
@@ -207,7 +207,7 @@ jobs:
     if: >-
       always() && (
         contains(needs.staging-repo.result, 'success') || (
-          contains(needs.*.result, 'skipped') && !needs.ci-preparation.outputs.may-deploy
+          contains(needs.staging-repo.result, 'skipped') && needs.ci-preparation.outputs.may-deploy == 'false'
```

Comment:
why?

Reply:
The same as before, it seems that outputs are evaluated as string.
https://docs.github.com/en/actions/learn-github-actions/expressions

The contains expression can be rolled back to "needs.*.result"